llama-index-core/llama_index/core/workflow/events.py

```diff
@@ -132,18 +140,30 @@ def __init__(self, result: Any = None) -> None:
         super().__init__(result=result)
 
 
-class InputRequiredEvent(Event):
-    """InputRequiredEvent is sent when an input is required for a step."""
+class UnblockingEvent(Event):
+    """A special event that unblocks the execution of Workflow run."""
 
-    prefix: str = Field(
-        description="The prefix and description of the input that is required."
-    )
 
+class BlockingEvent(Event):
+    """A special event that blocks the execution of Workflow run.
+
+    To unblock, the user will have to send an object of the class defined in
+    unblocking_event_type.
+    """
 
-class HumanResponseEvent(Event):
+    unblocking_event_type: ClassVar[Type[UnblockingEvent]]
```

Comment:
Why do we need to know the unblocking type? Shouldn't that already be known by the workflow

i.e. a step will emit one event, and accept another, with the assumption that something external is acting as the step

Reply:
Yea I think you're probably right it can be inferred by the workflow. That is, if you have a step that produces a blocking event then the next step's accepted event is the unblocking event? Guess I might be worried about some weird edge cases?

I make use of this coupling in my current PR for llama-deploy tho:

```
            async for ev in handler.stream_events():
                # check if blocking event
                if isinstance(ev, BlockingEvent):

                    # wait for the unblocking event type
                    logger.debug(f"Waiting for unblocking event: {ev}")
                    unblocking_ev = await self._events_buffer[current_call.task_id][
                        ev.unblocking_event_type
                    ].get()
```

Reply:
I think you don't even need to assume that -- the developer of the workflow could code it such that other work is done while waiting for the human response/unblocking event

Basically, the only purpose for having (Un)BlockingEvent is to indicate to the workflow to skip validation for those events, because the actual step that consumes/emits those events is external to the workflow

Reply:
That makes sense. I probably just need to re-think my implementation a little on the llama-deploy side.

Reply:
alright, removed coupling as its no longer needed on llama-deploy or in drawing :)